entity.go

```diff
@@ -391,22 +416,39 @@ func serializeStruct(src interface{}) ([]byte, error) {
 
 	se := getSerializationEncoder()
 	defer freeSerializationEncoder(se)
-	smd := &structMetaData{metaDatas: make([]string, 0, 16)}
-	_, fms := getFieldInfoAndMetadata(t) // Use this function to force generation if needed
-
-	if err := serializeStructInternal(se.enc, smd, fms, "", v); err != nil {
-		return nil, err
-	}
 
-	bufSize := se.buf.Len()
-	// final size = header + all metadatas + separators for metadata + data
-	finalBufSize := 1 + smd.totalLength + len(smd.metaDatas) + bufSize
-	finalBuf := make([]byte, finalBufSize)
-	finalBuf[0] = byte(serializationStateNormal)          // Set the header
-	serializeStructMetaData(finalBuf[1:], smd)            // Serialize the metadata
-	copy(finalBuf[finalBufSize-bufSize:], se.buf.Bytes()) // Copy the actual data
+	if ls, ok := src.(datastore.PropertyLoadSaver); ok {
+		se.buf.Write([]byte{serializationStatePropertyList})
+		props, err := ls.Save()
+		if err != nil {
+			return nil, err
+		}
+		for i := 0; i < len(props); i++ {
+			v := reflect.ValueOf(props[i].Value)
+			if v.Kind() == reflect.Ptr && v.IsNil() {
+				props[i].Value = nilType{}
```

Comment:
Use the new `nilValue` instead of allocating a new one with `nilType{}`.

Reply:
Fixed 0ffa7caaada1ace69f367ee36d201bcebed76bbd